File: integration-tests/src/test/java/dev/capylang/test/EnumDataFieldTest.java
package dev.capylang.test;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class EnumDataFieldTest {
    @Test
    void enumCanBeUsedAsDataFieldType() {
        var root = new EnumDataField.Path(EnumDataField.PathRoot.ABSOLUTE, List.of());
        var relative = new EnumDataField.Path(EnumDataField.PathRoot.RELATIVE, List.of("home"));

        assertThat(root.root()).isEqualTo(EnumDataField.PathRoot.ABSOLUTE);
        assertThat(relative.root()).isEqualTo(EnumDataField.PathRoot.RELATIVE);
        assertThat(root.segments()).isEmpty();
    }

    @Test
    void enumFieldCanBeMatchedAndRead() {
        var root = new EnumDataField.Path(EnumDataField.PathRoot.ABSOLUTE, List.of());
        var relative = new EnumDataField.Path(EnumDataField.PathRoot.RELATIVE, List.of());

        assertThat(EnumDataField.rootName(root)).isEqualTo("ABSOLUTE");
        assertThat(EnumDataField.isAbsolute(root)).isTrue();
        assertThat(EnumDataField.isAbsolute(relative)).isFalse();
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/EnumDataField.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('EnumDataField', () => {
    const enumData = generatedModule('dev/capylang/test/EnumDataField.js');
    const root = new enumData.Path({ root: enumData.ABSOLUTE, segments: [] });
    const relative = new enumData.Path({ root: enumData.RELATIVE, segments: [] });
    assert.equal(enumData.rootName(root), 'ABSOLUTE');
    assert.equal(enumData.isAbsolute(root), true);
    assert.equal(enumData.isAbsolute(relative), false);
});
